distribution/docker/build.gradle

```diff
@@ -11,39 +11,45 @@ testFixtures.useFixture()
 configurations {
   dockerPlugins
   dockerSource
+  ubiDockerSource
   ossDockerSource
+  ossUbiDockerSource
   restSpec
 }
 
 dependencies {
   dockerSource project(path: ":distribution:archives:linux-tar")
+  ubiDockerSource project(path: ":distribution:archives:linux-tar")
   ossDockerSource project(path: ":distribution:archives:oss-linux-tar")
+  ossUbiDockerSource project(path: ":distribution:archives:oss-linux-tar")
   restSpec project(':rest-api-spec')
 }
 
-ext.expansions = { oss, local ->
+ext.expansions = { oss, ubi, local ->
   final String classifier = 'linux-x86_64'            
   final String elasticsearch = oss ? "elasticsearch-oss-${VersionProperties.elasticsearch}-${classifier}.tar.gz" : "elasticsearch-${VersionProperties.elasticsearch}-${classifier}.tar.gz"
   return [
+    'base_image'          : ubi ? 'registry.access.redhat.com/ubi7/ubi-minimal:7.7' : 'centos:7',
     'build_date'          : project.ext.buildDate,
     'elasticsearch'       : elasticsearch,
     'git_revision'        : project.ext.gitRevision,
     'license'             : oss ? 'Apache-2.0' : 'Elastic-License',
+    'package_manager'     : ubi ? 'microdnf' : 'yum',
     'source_elasticsearch': local ? "COPY $elasticsearch /opt/" : "RUN cd /opt && curl --retry 8 -s -L -O https://artifacts.elastic.co/downloads/elasticsearch/${elasticsearch} && cd -",
     'version'             : VersionProperties.elasticsearch
   ]
 }
 
-private static String files(final boolean oss) {
-  return "build/${ oss ? 'oss-' : ''}docker"
+private static String files(final boolean oss, final boolean ubi) {
```

Comment:
I know this is in no way related to this PR, but can we rename this method. It's name clashes with `project.files()` which I find both confusing and makes it prone to error.

Reply:
I pushed 16a938d3b3dc8b5b9d33c569788c7fafbc836132.